kernel/userspace.c

```diff
@@ -92,6 +92,36 @@ const char *otype_to_str(enum k_objects otype)
 #endif
 }
 
+struct perm_ctx {
+	int parent_id;
+	int child_id;
+	struct k_thread *parent;
+};
+
+static void wordlist_cb(struct _k_object *ko, void *ctx_ptr)
+{
+	struct perm_ctx *ctx = (struct perm_ctx *)ctx_ptr;
+
+	if (sys_bitfield_test_bit((mem_addr_t)&ko->perms, ctx->parent_id &&
```

Comment:
should be:
if (sys_bitfield_test_bit((mem_addr_t)&ko->perms, ctx->parent_id) && (struct k_thread *)ko->name != ctx->parent)

Reply:
oh whoops, good catch